catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -213,4 +217,27 @@ public void processHeaders(Map<String, Object> headers, StorageRequest storageRe
                     .put(Constants.STORE_REFERENCE_KEY, ingestedFile.getAbsolutePath());
         }
     }
+
+    private void waitForAvailableSource(CatalogFramework catalogFramework)
+            throws SourceUnavailableException {
+        long startTime = System.currentTimeMillis();
+        boolean ready;
+        do {
+            if (timeoutHasBeenExceeded(startTime, 3 * 60 * 1000)) {
+                throw new SourceUnavailableException(
+                        "failed to find an available source within 3 minutes");
+            }
+
+            ready = catalogFramework.getSourceInfo(new SourceInfoRequestLocal(false))
+                    .getSourceInfo()
+                    .stream()
+                    .findFirst()
+                    .get()
+                    .isAvailable();
+        } while (!ready);
+    }
```

Comment:
An alternative implementation could use a `ScheduledExecutorService` instead of a sleep. That might be a bit more elegant.